File: 본문/python/Code_5_03_2.py
# 입력 → 배열 요소의 총합 S를 구함
N, K = map(int, input().split())
A = list(map(int, input().split()))
S = sum(A)

# 답 출력
if S % 2 != K % 2:
	print("No")
elif S > K:
	print("No")
else:
	print("Yes")

File: 본문/cpp/Code_5_03_2.cpp
#include <iostream>
using namespace std;

int N, K, A[59];
int sum = 0; // A[1] + A[2] + ... + A[N]의 값

int main() {
	// 입력 → 수열 요소의 총합 sum 구하기
	cin >> N >> K;
	for (int i = 1; i <= N; i++) cin >> A[i];
	for (int i = 1; i <= N; i++) sum += A[i];

	// 답 출력
	if (sum % 2 != K % 2) cout << "No" << endl;
	else if (sum > K) cout << "No" << endl;
	else cout << "Yes" << endl;
	return 0;
}